### DISCUSSION TITLE
Simple transactional HTTP transport?

### DISCUSSION THREAD

**Discussion by atesgoral** [modelcontextprotocol/.github#75] [General]
### Pre-submission Checklist

- [X] I have verified that this discussion would not be more appropriate as an issue in a specific repository
- [X] I have searched existing discussions to avoid duplicates

### Discussion Topic

It could be useful for the Inspector to support a simple transport that uses JSON-RPC over HTTP POST messages, for quicker integration/development against existing web servers.

Implementing the SSE + separate HTTP POST dance is a bit more involved and can be a speed bump for anxious early adopters.

What is the main motivation for the existence of the SSE + POST dance? Is there a drawback of relying on the response to the HTTP POST for the RPC invocation instead of relying on the response coming down the parallel SSE pipe?

*Update*: Yes, I see that the real-time notifications/updates wouldn't work, but a simple transport would be useful for a gradual integration.

**Comment by jspahrsummers** [comment]
I think this is a broader discussion than just the inspector, because it concerns the nature of the protocol and which transports are available. You may be interested in [this discussion on "stateless" MCP](https://github.com/modelcontextprotocol/specification/discussions/102). Would love your thoughts!

**Comment by fanpeeps** [comment]
How does a server announce which transport it uses (STDIO, which appears to be the default assumption, or SSE)?  Is a client meant to pull each computer communication method to determine?  Is server meant to broadcast (print) it’s transport method upon start up (before even receiving a request)?
